contracts/dao/neutron-chain-manager/src/msg.rs

```diff
@@ -0,0 +1,165 @@
+                let mut has_permission = false;
+                for permission in permissions {
+                    if let Permission::CronPermission(cron_permission) = permission {
+                        has_permission = cron_permission.add_schedule
+                    }
+                }
+
+                has_permission
+            }
+        }
+    }
+    pub fn has_cron_remove_schedule_permission(&self) -> bool {
+        match self {
+            Strategy::AllowAll => true,
+            Strategy::AllowOnly(permissions) => {
+                let mut has_permission = false;
+                for permission in permissions {
+                    if let Permission::CronPermission(cron_permission) = permission {
+                        has_permission = cron_permission.remove_schedule
+                    }
+                }
+
+                has_permission
+            }
+        }
+    }
+    pub fn has_param_change_permission(&self, param_change: ParamChange) -> bool {
+        match self {
+            Strategy::AllowAll => true,
+            Strategy::AllowOnly(permissions) => {
+                for permission in permissions {
+                    if let Permission::ParamChangePermission(param_change_permissions) = permission
+                    {
+                        for param_change_permission in param_change_permissions.params.clone() {
+                            if param_change.subspace == param_change_permission.subspace
+                                && param_change.key == param_change_permission.key
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                false
+            }
+        }
+    }
+
+    pub fn get_cron_update_param_permission(&self) -> Option<CronUpdateParamsPermission> {
+        match self {
+            Strategy::AllowAll => Some(CronUpdateParamsPermission {
+                security_address: true,
+                limit: true,
+            }),
+            Strategy::AllowOnly(permissions) => {
+                for permission in permissions {
+                    if let Permission::UpdateParamsPermission(update_params_permission) = permission
+                    {
+                        return match update_params_permission {
+                            UpdateParamsPermission::CronUpdateParamsPermission(
+                                cron_update_params,
+                            ) => Some(cron_update_params.clone()),
+                        };
+                    }
+                }
+
+                None
+            }
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub enum Permission {
+    // Deprecated, for legacy parameter updates using `params` module.
+    ParamChangePermission(ParamChangePermission),
+    UpdateParamsPermission(UpdateParamsPermission),
```

Comment:
Pls add a comment which describes the difference between two sets of permissions.